doc/user_guide/setup_dot_py.rst

```diff
@@ -12,11 +12,6 @@ packages define the installation files in a file called ``setup.py``
 in the project root. The setup.py file uses Python to describe the
 Python content of the stack.
 
-We recommend to prefer distutils package over setuptools/distribute,
-because with distutils we can avoid the creation of egg-info folders
```

Comment:
Does this limitation still apply with `setuptools`?

Reply:
@sloretz Friendly ping.

Reply:
Maybe not? I built all Noetic repos with source entries using Buster. The only three using `setuptools` in my workspace are `catkin`, `genmsg`, and `python_qt_binding`. After building with this branch all three have clean source spaces, and `find .-name "*egg*"` in `src` shows no files. I assume that means whatever avoids `egg-info` folder creation was able to do so.

Reply:
`egg-info` folders are still created when building Melodic packages w/ python 2 on Bionic (See https://travis-ci.org/github/ros-controls/ros_control/jobs/661982516?utm_medium=notification&utm_source=github_status)

Is this expected?

Reply:
That's weird that it tried to create it in the source. In all my testing, the `egg-info` folder always ends up in the installspace.

Reply:
I see the same thing when building locally as well, for all packages that have been migrated to `setuptools`. I tested `angles`, `ros_control`, and `catkin` itself.

Reply:
I don't think that's expected, and I also wouldn't expect the behavior to have been changed by this PR. [`controller_manager_msgs` is still using `distutils` on `melodic-devel`](https://github.com/ros-controls/ros_control/blob/f211ee381d4f40967db980601e9a94ac40f3c8c0/controller_manager_msgs/setup.py#L3).

Reply:
@matthew-reynolds maybe [`python_distutils_install.sh`](https://github.com/ros/catkin/blob/86439ec5d2010d5c47c30b815aa97e525037930f/cmake/templates/python_distutils_install.sh.in#L24-L32) needs to set [`--egg-base`](https://setuptools.readthedocs.io/en/latest/setuptools.html#other-egg-info-options) ? I'm not sure that option is also available in `distutils` though.

Reply:
The svem part is no longer relevant, but if you need to resuscitate the notion of setuptools-specific flags, the pieces of this diff that detect it may be useful:

https://github.com/ros/catkin/compare/noetic-devel...mikepurvis:add-setuptools-svem

Reply:
@matthew-reynolds would you mind opening an issue with steps to reproduce? I'm unable to reproduce using ROS Melodic on Ubuntu Bionic with :

```
cd /tmp
mkdir -p mel_ws/src
cd mel_ws/src
git clone git@github.com:ros/angles.git
chmod a-w -R angles
cd /tmp/mel_ws
catkin_make_isolated
catkin_make
```

I was expecting `catkin_make_isolated` or `catkin_make` to fail, but they both succeeded. If they tried to create `egg_info` folders in the source directory then they should get a permission denied error since it's read-only.

Reply:
Interesting - I did a little more testing, this seems to only occur when installing.

`catkin_make install`, `catkin_make_isolated --install`, and `catkin config --install; catkin build` all fail.

```
cd /tmp
mkdir -p mel_ws/src
cd mel_ws/src
git clone git@github.com:ros/angles.git
chmod a-w -R angles
cd /tmp/mel_ws
catkin_make install                                 # Fails
catkin_make_isolated --install                      # Fails
catkin clean; catkin config --install; catkin build # Fails
```

I will open an issue to continue this conversation :+1:

Reply:
#1069